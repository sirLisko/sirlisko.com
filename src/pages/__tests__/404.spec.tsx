import React from "react";
import { render } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import Page404 from "../404";

jest.mock("gatsby", () => ({
  useStaticQuery: jest.fn(),
  graphql: jest.fn(),
}));
const mockedStaticQuery = useStaticQuery as jest.Mock;

beforeEach(() => {
  mockedStaticQuery.mockImplementationOnce(() => ({
    site: {
      siteMetadata: {
        title: `I am the title`,
      },
    },
  }));
});

describe("404 Page", () => {
  it("should render properly", () => {
    const { asFragment } = render(<Page404 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
