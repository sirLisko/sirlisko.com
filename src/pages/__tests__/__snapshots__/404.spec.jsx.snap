// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 Page should render properly 1`] = `
<DocumentFragment>
  <h1>
    NOT FOUND
  </h1>
  <p>
    You just hit a route that doesn't exist... the sadness.
  </p>
</DocumentFragment>
`;
